packages/connect-disputable-voting/src/models/Vote.ts

```diff
@@ -172,6 +177,46 @@ export default class Vote {
     return wasAcceptedBeforeLastFlip != this.isAccepted
   }
 
+  async hasEndedExecutionDelay(): Promise<boolean> {
+    const setting = await this.setting()
+    const currentTimestamp = currentTimestampEvm()
+    const executionDelayEndDate = bn(this.endDate).add(setting.executionDelay)
+    return currentTimestamp.gte(executionDelayEndDate)
+  }
+
+  async canVote(voterAddress: Address): Promise<boolean> {
+    return !this.hasEnded &&
+      this.voteStatus === 'Scheduled' &&
+      !(await this.hasVoted(voterAddress)) &&
+      (await this.votingPower(voterAddress)).gt(bn(0))
+  }
+
+  async canExecute(): Promise<boolean> {
+    return this.isAccepted &&
+      this.voteStatus === 'Scheduled' &&
+      (await this.hasEndedExecutionDelay())
+  }
+
+  async votingPower(voterAddress: Address): Promise<BigNumber> {
+    const token = await this.token()
+    return token.getBalanceAt(voterAddress, this.snapshotBlock)
+  }
+
+  async formattedVotingPower(voterAddress: Address): Promise<string> {
+    const token = await this.token()
+    const balance = await token.getBalanceAt(voterAddress, this.snapshotBlock)
+    return formatBn(balance, token.decimals)
+  }
+
+  async token(): Promise<ERC20> {
+    return this.#connector.ERC20(this.tokenId)
+  }
+
+  async hasVoted(voterAddress: Address): Promise<boolean> {
+    const castVote = await this.castVote(voterAddress)
+    return castVote !== null
+  }
+
```

Comment:
Not necessarily for this PR, but we should probably have subscription equivalents for these methods, apart from `token()`.

Reply:
I'm starting to think we should find a more automated way to have these haha

Reply:
Definitely 😬